modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/BaseDDMStructureLayoutUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,143 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public abstract class BaseDDMStructureLayoutUpgradeProcessTestCase
+	extends BaseCTUpgradeProcessTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), TestPropsValues.getUserId());
+	}
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString());
+
+		DDMFormLayout ddmFormLayout = createDDMFormLayoutPaginationMode();
+
+		DDMStructure ddmStructure = _ddmStructureLocalService.addStructure(
+			TestPropsValues.getUserId(), _group.getGroupId(), 0,
+			PortalUtil.getClassNameId(DDMFormInstance.class.getName()), null,
+			HashMapBuilder.put(
+				LocaleUtil.getSiteDefault(), "Test Structure"
+			).build(),
+			null, ddmForm, ddmFormLayout, StorageType.DEFAULT.toString(),
+			DDMStructureConstants.TYPE_DEFAULT, _serviceContext);
+
+		return ddmStructure.fetchDDMStructureLayout();
+	}
+
+	protected DDMFormLayout createDDMFormLayoutPaginationMode() {
+		DDMFormLayout ddmFormLayout = DDMUtil.getDefaultDDMFormLayout(
+			DDMFormTestUtil.createDDMForm(
+				RandomTestUtil.randomString(), RandomTestUtil.randomString()));
+
+		ddmFormLayout.setPaginationMode(getPaginationMode());
+
+		return ddmFormLayout;
+	}
+
+	@Override
+	protected CTService<?> getCTService() {
+		return _ddmStructureLayoutLocalService;
+	}
+
+	protected abstract String getPaginationMode();
+
+	protected abstract String getSchemaVersion();
+
+	@Override
+	protected void runUpgrade() throws Exception {
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator,
+			"com.liferay.dynamic.data.mapping.internal.upgrade." +
+				getSchemaVersion() + ".DDMStructureLayoutUpgradeProcess");
+
+		upgradeProcess.upgrade();
+	}
+
+	@Override
+	protected CTModel<?> updateCTModel(CTModel<?> ctModel) throws Exception {
+		DDMStructureLayout ddmStructureLayout = (DDMStructureLayout)ctModel;
+
+		DDMFormLayout ddmFormLayout = createDDMFormLayoutPaginationMode();
+
+		return _ddmStructureLayoutLocalService.updateStructureLayout(
+			ddmStructureLayout.getStructureLayoutId(), ddmFormLayout,
```

Comment:
@p-albuquerque please inline here

```
DDMStructureLayout ddmStructureLayout = (DDMStructureLayout)ctModel;

return _ddmStructureLayoutLocalService.updateStructureLayout(
	ddmStructureLayout.getStructureLayoutId(),
	_createDDMFormLayout(DDMFormTestUtil.createDDMForm()),
	_serviceContext);
```

Reply:
@carolmariaabb Leave createDDMForm without a **fieldName** cause the error:
![fieldError](https://github.com/user-attachments/assets/5912f34d-28fd-4dd4-9181-9d1001f6011e)

Can I proceed this way?
```java
		DDMStructureLayout ddmStructureLayout = (DDMStructureLayout)ctModel;

		return _ddmStructureLayoutLocalService.updateStructureLayout(
			ddmStructureLayout.getStructureLayoutId(),
			_createDDMFormLayout(
				DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString())),
			_serviceContext);
```

Reply:
@p-albuquerque yes